driver/pelican/demandResponse.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"encoding/xml"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/immesys/spawnpoint/spawnable"
+	bw2 "gopkg.in/immesys/bw2bind.v5"
+)
+
+type ADREventWrapperAPI struct {
+	XMLName xml.Name    `xml:"result"`
+	Success int         `xml:"success"`
+	Message string      `xml:"message"`
+	Event   ADREventAPI `xml:"attribute"`
+}
+
+type ADREventAPI struct {
+	End    string `xml:"OpenADREventEnd"`
+	Start  string `xml:"OpenADREventStart"`
+	Status string `xml:"OpenADRStatus"`
+	Type   string `xml:"OpenADREventType"`
+}
+
+type ADREvent struct {
+	Event_End   int64
+	Event_Start int64
+	Event_Type  int64
+	DR_Status   int64
+	Time        int64
+}
+
+func (pel *Pelican) TrackDREvent() (*ADREvent, error) {
+	resp, _, errs := pel.req.Get(pel.target).
+		Param("username", pel.username).
+		Param("password", pel.password).
+		Param("request", "get").
+		Param("object", "Site").
+		Param("value", "OpenADREventEnd;OpenADREventStart;OpenADRStatus;OpenADREventType").
+		End()
+
+	if errs != nil {
+		return nil, fmt.Errorf("Error retrieving thermostat status from %s: %v", pel.target, errs)
```

Comment:
Let's make this error message more specific, we're trying to retrieve information about a DR event.